flink-core/src/main/java/org/apache/flink/configuration/RestartBackoffTimeStrategyOptions.java

```diff
@@ -25,6 +25,16 @@
  */
 @PublicEvolving
 public class RestartBackoffTimeStrategyOptions {
+
+	/**
+	 * Class name of the RestartBackoffTimeStrategy implementation to use.
+	 */
+	@PublicEvolving
+	public static final ConfigOption<String> RESTART_BACKOFF_TIME_STRATEGY_CLASS_NAME = ConfigOptions
+		.key("restart-backoff-time-strategy.class-name")
```

Comment:
With the introduction of the new scheduler, users will not be able to use custom implementations of the `RestartStrategy` interface. Strictly speaking this is a breaking change. However, it is not clear to me if there are any Flink users at all that require a custom `RestartStrategy`. I think it makes sense to start a discussion on the dev mailing list. Until it is clear that users require this level of customization, we should not offer users to write their own back off time strategy for now.

Reply:
Yes it is a breaking change if a user was using a custom RestartStrategy.
I will open a discussion to
1. notify users that previous custom RestartStrategy does not work in the new scheduler anymore
2. discuss whether we should keep supporting custom RestartBackoffTimeStrategy for existing custom RestartStrategy migration or future use

Reply:
The survey is concluded that we do not support restart strategy customization. 
I will also remove the description of restart strategy customization in a hotfix commit, so that we do not make it a public interface.